### PR TITLE
Reapply "Upgrade to Debian Bookworm and Python 3.11 (#4028)"

### PR DESCRIPTION
**Pull request by past**
This is reapplying the changes from #4028 while fixing the selenium issues.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,20 +1,31 @@
...
-FROM golang:1.22.7-bullseye
+FROM golang:1.22.7-bookworm
 
 # Create a non-priviledged user to run browsers as (Firefox and Chrome do not
 # like to run as root).
 RUN chmod a+rx $HOME && useradd --uid 9999 --user-group --create-home browser
 
+# Add apt repositories for Java, Node.js and Google Cloud CLI
+RUN export DISTRO_CODENAME=$(awk -F= '/^VERSION_CODENAME/{print$2}' /etc/os-release) && \
+    echo "deb [signed-by=/usr/share/keyrings/correto.gpg] https://apt.corretto.aws stable main" > /etc/apt/sources.list.d/adoptium.list && \
```
Comment (jcscottiii) [inline Dockerfile:10]: Nit: I see adoptium here. I don't see it in the [instructions](https://docs.aws.amazon.com/corretto/latest/corretto-11-ug/generic-linux-install.html#debian-install-instruct). Let's change it to correcto
```suggestion
    echo "deb [signed-by=/usr/share/keyrings/correto.gpg] https://apt.corretto.aws stable main" > /etc/apt/sources.list.d/corretto.list && \
```

File: Makefile [hunk 2]
```diff
@@ -229,15 +229,15 @@ gcc: apt-get-gcc
...
+python3: apt-get-python3.11
 tox: apt-get-tox
 unzip: apt-get-unzip
 wget: apt-get-wget
 
 java:
 	@ # java has a different apt-get package name.
 	if [[ "$$(which java)" == "" ]]; then \
-		sudo apt-get install -qqy --no-install-suggests openjdk-11-jdk; \
+		sudo apt-get install -qqy --no-install-suggests temurin-11-jdk; \
```
Comment (jcscottiii) [inline Makefile:240]: This should match the package installed in the Dockerfile.
```suggestion
		sudo apt-get install -qqy --no-install-suggests java-11-amazon-corretto-jdk; \
```

**Review by jcscottiii — APPROVED** [review]
LGTM with two nits. The nits are to make sure we are using either [correcto](https://docs.aws.amazon.com/corretto/latest/corretto-11-ug/generic-linux-install.html#debian-install-instruct) or [adoptium](https://adoptium.net/) consistently. (In this case, correcto)

File: Dockerfile (hunk 1 above)
Comment (past) [reply Dockerfile]: Oops, good catch. I also fixed the typo in the keyring file name.

File: Makefile (hunk 2 above)
Comment (past) [reply Makefile]: Thanks for spotting this! Fixed.

**Review by jcscottiii — APPROVED** [review]

**Pull request merged**